pinot-spi/src/test/java/org/apache/pinot/spi/env/VersionedPropertyConfigTest.java

```diff
@@ -0,0 +1,224 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.env;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import org.apache.commons.configuration2.PropertiesConfiguration;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+
+
+public class VersionedPropertyConfigTest {
+
+  private static final String COLON_SEPARATOR = ":";
+  private static final File TEMP_DIR = new File(FileUtils.getTempDirectory(), "VersionedPropertyConfigTest");
+  private static final File CONFIG_FILE = new File(TEMP_DIR, "config");
+  private static final String[] TEST_PROPERTY_KEY = { "test1", "test2_key", "test3_key_",
+      "test4_key_1234", "test-1", "test.1" };
+  private static final String[] TEST_PROPERTY_KEY_WITH_SPECIAL_CHAR = { "test:1", "test2=key",
```

Comment:
curious how keys with `=` like `test2=key` get handled properly? because the two white spaces in the separator constant defined in `CommonsConfigurationUtils.java` above?

Reply:
So, in this test case, `test2=key` is a key. the overall property in the file looks like this
```bash
test2=key = value.
```
Since we are using the `getPropertySeparator` method in the reader, which returns the separator value as ` = `, it's working correctly. I think with this implementation, we can also use  `=` in the key.